packages/core/useVirtualList/index.test.ts

```diff
@@ -1,11 +1,21 @@
-import { describe, expect, it } from 'vitest'
+import { describe, expect, it, vi } from 'vitest'
 import { ref as deepRef } from 'vue'
 import { useVirtualList } from './index'
 
 describe('useVirtualList', () => {
   it('should be defined', () => {
     expect(useVirtualList).toBeDefined()
   })
+
+  it('should not warn when list is a plain array', async () => {
+    const spy = vi.spyOn(console, 'warn').mockImplementation(() => {})
+
+    useVirtualList(['a', 'b', 'c', 'd', 'e', 'f'], { itemWidth: () => 50, overscan: 1 })
+
+    expect(spy).not.toHaveBeenCalled()
```

Comment:
i dont think this'll ever be called since we don't console.warn in there

either way, we should test that mutations are observed instead i think

something like this:

```ts
    const items = shallowRef(['a', 'b'])
    const {
      list,
      containerProps: { ref: containerRef },
      scrollTo,
    } = useVirtualList(items, { itemHeight: () => 50 })
    const div = { ...document.createElement('div'), clientHeight: 100 }

    containerRef.value = div
    scrollTo(0)

    items.value = ['c', 'd']
    await nextTick()

    expect(list.value.map(i => i.data)).toEqual(['c', 'd'])

```